### PR TITLE
New multiselect

### PR DESCRIPTION
**Pull request by pstjean**
This switches over to use the ng-select library. The tests pass, and QA in the demo app works, but I have fairly low confidence that this will actually work in production. A first look would be welcome.

### CODE REVIEW

**Comment by pstjean** [comment]
Testing this morning by linking the new Styleguide into Publish. The new library doesn't survive contact with reality, as I feared, so I'll start debugging.

**Comment by pstjean** [comment]
I've done some significant testing using the new library in Publish locally and making modifications to fix resulting issues.

I've removed two elements of functionality from the new version, because they appear unnecessary now:
1. No longer sorting options manually, as I didn't observe ng-select doing any sort of arbitrary reordering.
2. No longer converting options passed to `prx-select` with `single` set to `true` into an array with one element, as `ng-select` expects the format we were already passing (`any` for single options, `any[]` for multiselect).

File: projects/ngx-prx-styleguide/src/lib/select/selected.component.html [hunk 1]
```diff
@@ -0,0 +1,12 @@
+<ng-select
```
Comment (sandikbarr) [inline projects/ngx-prx-styleguide/src/lib/select/selected.component.html]: nit: should this file be named `select.component.html` instead of "selected"?

**Comment by pstjean** [comment]
Because of the way `selected` is bound into and out of the component, I ended up with yet another browser tab destroying infinite loop of change detection when attempting to filter items out of `selected`. This new approach uses different properties to track selected items internal to the component (`ngSelectSelected`) and avoids modifying `selected`, which preserves the boundary between our component and the outside world better.

I also observed behavior where the component will receive `selected` on a separate (and earlier) change detection tick than it receives `options`, which makes filtering just clobber all selections, so there's some logic to look at the changesets as we receive them and avoid filtering until after we're sure we've got some options to work with.

File: projects/ngx-prx-styleguide/src/lib/select/select.component.ts [hunk 2]
```diff
@@ -45,77 +32,26 @@ export class SelectComponent {
...
 
-  private orderSelected() {
-    let ids = this.msOptions.map(o => o.id);
-    this._selected = this._selected.sort((a, b) => {
-      return ids.indexOf(a) - ids.indexOf(b);
+  private convertOptions(): NgOption[] {
+    return this.options.map(opt => {
+      if (opt instanceof Array) {
+        return {name: opt[0], value: opt[1]};
+      } else {
```
[the remark this answers is not in the document]
Comment (pstjean) [reply projects/ngx-prx-styleguide/src/lib/select/select.component.ts]: We always receive the options from the parent component in the format `any[] | any[][]`, so this conversion is always needed.

[the remark this answers is not in the document]
Comment (rpeterman-gp) [reply projects/ngx-prx-styleguide/src/lib/select/select.component.ts]: nvm. this seems out of scope for this ticket.

**Pull request merged**